src/build123d/direct_api.py

```diff
@@ -3618,7 +3619,7 @@ def project_text(
 
 
 # This TypeVar allows IDEs to see the type of objects within the ShapeList
-T = TypeVar("T", bound=Shape)
+T = TypeVar("T", Shape, Vector)
```

Comment:
This is all magic to me, it was better than nothing but didn't work perfectly - hopefully this resolves it.